### ISSUE TITLE
HTTPS support

### ISSUE THREAD

**Issue by SudoCat** [ara-framework/nova-proxy#9]
It doesn't seem to be possible to use this proxy with a HTTPS/SSL host - is that correct?

**Comment by marconi1992** [comment]
That's correct.  We're not supporting hosts with HTTPS/SSL mainly because `Nova Proxy` was planned to be used as a sidecar proxy, then we can use https in outer layers e.g a load balancer.

**Comment by marconi1992** [comment]
good catch!  @SudoCat, we'll elaborate more that explanation and update the Readme file accordingly.

**Comment by SudoCat** [comment]
Thanks for the lightning fast response! I must admit that this sort of thing is far outside my area of expertise, but we're hoping to use the ara framework to help us slowly migrate our front end away from a monolithic java ecommerce platform; I'm worried this limitation might prove to be an impassable blocker for our use case. 

Would there be any plans to add SSL support to the proxy layer, or would we need to consider forking/utilising an alternative?

**Comment by marconi1992** [comment]
Unfortunately, we don't have plans to add SSL support. We guess decoding the response to parse the HTML, as well as, coding it back could add performance issues. However, as I said it's only I guess. Let's keep open this issue and we'll make a proof of concept to see what we get.

**Comment by zeenotheinventor** [comment]
I'm interested in this as well. Would you be okay with me raising a PR to add this functionality?

**Comment by marconi1992** [comment]
Sure @zeenotheinventor feel free to do it, I'll really appreciate it.
